docs/content/guides/dagster-pipes/aws-lambda.mdx

```diff
@@ -0,0 +1,288 @@
+  way.
+</Note>
+
+1. In the **Code source** section of the page, add a new file. This can be accomplished with **File > New file** or by clicking the green **+** icon next to the open `lambda_function` tab:
+
+   <Image
+   alt="Highlighted New file option in a menu of the Code source section on the Lambda function details page"
+   src="/images/guides/dagster-pipes/aws-lambda/aws-lambda-create-new-file.png"
+   width={2386}
+   height={1196}
+   />
+
+2. **In a new browser tab**, navigate to the following URL:
+
+   ```shell
+   https://raw.githubusercontent.com/dagster-io/dagster/master/python_modules/dagster-pipes/dagster_pipes/__init__.py
+   ```
+
+3. Copy the contents of `__init__.py` into the new file you created in AWS. **Note**: We recommend adding the source link and the date you copied the contents to the top of the file as comments:
+
+   <Image
+   alt="The copied contents of the Dagster Pipes file into a file in the AWS UI"
+   src="/images/guides/dagster-pipes/aws-lambda/aws-lambda-add-dagster-pipes.png"
+   width={2390}
+   height={1090}
+   />
+
+4. Save the file as `dagster_pipes.py`.
+
+### Step 1.3: Add the code to execute to the function
+
+In this step, you'll add the code you want to execute to the function. Create another file in the AWS UI - or use the default `lambda_function.py` file created by the function - and paste in the following code:
+
+<!-- TODO: Confirm this is actually correct -->
+
+```python
+from dagster_pipes import PipesMappingParamsLoader, open_dagster_pipes
+
+
+def lambda_handler(event, _context):
+
+    with open_dagster_pipes(
+        params_loader=PipesMappingParamsLoader(event),
+    ) as pipes:
+        # Stream log message back to Dagster
+        pipes.log.info(f"Using some_parameter value: {some_parameter_value}")
+
+        # ... your code that computes and persists the asset
+
+        # Stream asset materialization metadata and data version back to Dagster.
+        # This should be called after you've computed and stored the asset value. We
+        # omit the asset key here because there is only one asset in scope, but for
+        # multi-assets you can pass an `asset_key` parameter.
+
+        pipes.report_asset_materialization(
+            metadata={
+                "some_metric": {"raw_value": some_parameter_value + 1, "type": "int"}
+            },
+            data_version="alpha",
+        )
+```
+
+Let's review what this code does:
+
+- Imports `PipesMappingParamsLoader` and <PyObject object="open_dagster_pipes" module="dagster_pipes" /> from `dagster_pipes`
+
+- **Initializes the Dagster Pipes context (<PyObject object="open_dagster_pipes" module="dagster_pipes" />), which yields an instance of <PyObject object="PipesContext" module="dagster_pipes" /> called `pipes`.**
+
+  <!-- TODO: Looks like this paragraph will need to be updated a bit, based on the fact that we're using PipesMappingParamsLoader -->
```

Comment:
I could use some help updating this part. I think I remember you saying that `PipesMappingParamsLoader` is what loads context here, but I'm still a bit fuzzy on what this `event` argument is and how it works.

Reply:
Ya this will be something like: "The PipesLambdaClient injects the information needed for pipes in to the event payload. To establish the pipes correctly on the lambda side, we will need to pass the payload to `PipesMappingParamsLoader` and use that in `open_dagster_pipes` "